magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordPersistence.scala

```diff
@@ -391,62 +410,62 @@ object DefaultRecordPersistence extends Protocols with DiffsonProtocol with Reco
 
         JsonDiff.diff(oldAspectJson, newAspectJson, false)
       }
-      result <- patchRecordAspectById(session, recordId, aspectId, recordAspectPatch)
+      result <- patchRecordAspectById(session, recordId, tenantId, aspectId, recordAspectPatch)
     } yield result
   }
 
-  def createRecord(implicit session: DBSession, record: Record): Try[Record] = {
+  def createRecord(implicit session: DBSession, record: Record, tenantId: BigInt): Try[Record] = {
     for {
       eventId <- Try {
-        val eventJson = CreateRecordEvent(record.id, record.name).toJson.compactPrint
-        sql"insert into Events (eventTypeId, userId, data) values (${CreateRecordEvent.Id}, 0, $eventJson::json)".updateAndReturnGeneratedKey.apply()
+        val eventJson = CreateRecordEvent(record.id, tenantId, record.name).toJson.compactPrint
+        sql"insert into Events (eventTypeId, userId, tenantId, data) values (${CreateRecordEvent.Id}, 0, $tenantId, $eventJson::json)".updateAndReturnGeneratedKey.apply()
       }
       insertResult <- Try {
-        sql"""insert into Records (recordId, name, lastUpdate, sourcetag) values (${record.id}, ${record.name}, $eventId, ${record.sourceTag})""".update.apply()
+        sql"""insert into Records (recordId, tenantId, name, lastUpdate, sourcetag) values (${record.id}, $tenantId, ${record.name}, $eventId, ${record.sourceTag})""".update.apply()
       } match {
         case Failure(e: SQLException) if e.getSQLState().substring(0, 2) == "23" =>
           Failure(new RuntimeException(s"Cannot create record '${record.id}' because a record with that ID already exists."))
         case anythingElse => anythingElse
       }
 
-      hasAspectFailure <- record.aspects.map(aspect => createRecordAspect(session, record.id, aspect._1, aspect._2)).find(_.isFailure) match {
+      hasAspectFailure <- record.aspects.map(aspect => createRecordAspect(session, record.id, tenantId, aspect._1, aspect._2)).find(_.isFailure) match {
         case Some(Failure(e)) => Failure(e)
         case _                => Success(record)
       }
     } yield hasAspectFailure
   }
 
-  def deleteRecord(implicit session: DBSession, recordId: String): Try[Boolean] = {
+  def deleteRecord(implicit session: DBSession, recordId: String, tenantId: BigInt): Try[Boolean] = {
     for {
       aspects <- Try {
-        sql"select aspectId from RecordAspects where recordId=$recordId"
+        sql"select aspectId from RecordAspects where (recordId, tenantId)=($recordId, $tenantId)"
```

Comment:
Is this more performant (or at least _as_ performant) as `where recordId=$recordId and tenantId=$tenantId`?

Reply:
No evidence of improvement. Maybe just cosmetic?